src/shared/sleep-util.c

```diff
@@ -1028,6 +1028,45 @@ int read_fiemap(int fd, struct fiemap **ret) {
         return 0;
 }
 
+int write_resume_config(dev_t devno, uint64_t offset, const char *device) {
+        char offset_str[DECIMAL_STR_MAX(uint64_t)];
+        _cleanup_free_ char *path = NULL;
+        const char *devno_str;
+        int r;
+
+        devno_str = FORMAT_DEVNUM(devno);
+        xsprintf(offset_str, "%" PRIu64, offset);
+
+        if (!device) {
+                r = device_path_make_canonical(S_IFBLK, devno, &path);
+                if (r < 0)
+                        return log_oom();
+                device = path;
+        }
+
+        /* We write the offset first since it's safer. Note that this file is only available in 4.17+, so
+         * fail gracefully if it doesn't exist. */
+        r = write_string_file("/sys/power/resume_offset", offset_str, WRITE_STRING_FILE_DISABLE_BUFFER);
+        if (r == -ENOENT)
+                log_warning_errno(r, "/sys/power/resume_offset is unavailable, skipping writing swap file offset.");
```

Comment:
hmm, i think we should make this safer by returning a fatal error if the sysfs attribute for this does not exist, and the offset is non-zero.

i.e.:

```c
if (r == -ENOENT) {
       if (offset != 0)
               return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Can't configure hibernation offset %" PRIu64, kernel does not support /sys/power/resume_offset.");
       log_warning_errno(r, "…);
}
…
```

you get the idea